packages/account/src/Sections/Profile/LanguageSettings/language-settings.tsx

```diff
@@ -1,68 +1,43 @@
-                return (
-                    <form onSubmit={handleSubmit} className='account-form account-form--language-settings'>
-                        <div className='settings-language'>
-                            <DesktopWrapper>
-                                <FormSubHeader title={localize('Select Language')} />
-                            </DesktopWrapper>
-                            <div className='settings-language__language-container'>
-                                {allowed_language_keys.map(language_key => {
-                                    return (
-                                        <LanguageRadioButton
-                                            key={language_key}
-                                            id={language_key}
-                                            language_code={language_key}
-                                            is_current_language={values.language_code === language_key}
-                                            name='language-radio-group'
-                                            onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
-                                                setFieldValue('language_code', event.target.value)
-                                            }
-                                        />
-                                    );
-                                })}
-                            </div>
-                        </div>
-                        <FormFooter>
-                            <Button
-                                className='account-form__footer-btn'
-                                type='submit'
-                                data-testid={'submit-button'}
-                                has_effect
-                                text={localize('Submit')}
-                                large
-                                primary
-                                is_disabled={isCurrentLanguage(values.language_code)}
-                            />
-                        </FormFooter>
-                    </form>
-                );
-            }}
-        </Formik>
+        <div className='settings-language'>
+            <FormSubHeader title={localize('Select Language')} />
+            <div className='settings-language__language-container'>
+                {allowed_language_keys.map(language_key => {
+                    return (
+                        <LanguageRadioButton
+                            key={language_key}
+                            id={language_key}
+                            language_code={language_key}
+                            is_current_language={current_language === language_key}
+                            name='language-radio-group'
+                            onChange={async () => {
+                                await changeSelectedLanguage(language_key);
+                                await i18n.changeLanguage?.(language_key);
```

Comment:
@yauheni-deriv Is there a reason for awaiting here? 🤔

Reply:
The above functions returns a Promise

Reply:
@likhith-deriv Yes but do we need to wait for `changeSelectedLanguage` to finish and then change the language for `i18n`? 🤔
By the way, seems like `changeSelectedLanguage` will call `i18n.changeLanguage` internally, So I think we can remove it, What do you think? 🤔

Reply:
Didn’t notice this comment :( , I think we can leave it as it is for now.
it had been implemented before and the card has been tested so far.

Reply:
will refactor it :)